### PR TITLE
Implement a modal with mentor/mentee details in ScholarX Archive page

### LINKED ISSUE #1441: Implement a modal with mentor/mentee details in ScholarX Archive page
**Is your feature request related to a problem? Please describe.**
Currently, on the ScholarX Archive page, mentor/ mentee details are not displayed.
You can visit the page at [https://sefglobal.org/scholarx/archive/](https://sefglobal.org/scholarx/archive/).

**Describe the solution you'd like**
Once a viewer clicks a profile of a mentor/mentee in the archive, a modal should appear with all the details of that mentor/ mentee. 
This can be similar to the one we have in the [https://sefglobal.org/scholarx/past-programs/2021/](https://sefglobal.org/scholarx/past-programs/2021/) page.
Screenshot 2022-11-28 at 13 48 11

**Describe alternatives you've considered**
None

**Additional context**
All the details are available in the [API](https://script.google.com/macros/s/AKfycbw0TVyldiK5ijUxjLJkhrxHpxZpIjoeLytZqrvOlftYeHywn1hBEnL0aHGIS8hOFxPp/exec) and have already used in the [https://sefglobal.org/scholarx/archive/](https://sefglobal.org/scholarx/archive/) page.

The following details should be displayed.

**For the mentors:**
- Name
- Title
- Image (hyperlinked to the LinkedIn profile)
- Bio
- Areas of expertise
- Projects
- Mentees[]

**For the mentees:**
- Name
- Image (hyperlinked to the LinkedIn profile)
- Mentor's name
- University

**Are you working on it?**
No

### PR DESCRIPTION
**Pull request by YoshithaRathnayake**
## Purpose

The purpose of this PR is to fix #1441 

## Goals

Implementing a modal with mentor/mentee details in ScholarX Archive page

## Approach

Implemented a modal with mentor/mentee details in ScholarX Archive page

### Screenshots

image
  
### Preview Link

https://pr-1459-sef-site.surge.sh/scholarx/archive

##  Checklist
- [x] I have read and understood the [development best practices](https://handbook.sefglobal.org/organisation/engineering-team#development-best-practices) guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Madhawa97** [comment]
it's a bit hard to identify what you have actually changed in the code @YoshithaRathnayake. Do you use any code formatting extensions like Prettier?

**Comment by YoshithaRathnayake** [comment]
> it's a bit hard to identify what you have actually changed in the code @YoshithaRathnayake. Do you use any code formatting extensions like Prettier?

Yeah some because some codes are not aligned correctly, I used SHIFT+ALT+F in vs code to format the code

File: scholarx/archive/functions.js [hunk 1]
```diff
@@ -58,3 +58,22 @@ async function loadData() {
     $("#menteeProfiles").html(mentee_Profiles);
 }
 loadData();
+
```
Comment (YoshithaRathnayake) [inline scholarx/archive/functions.js:75]: Added from this line

File: scholarx/archive/index.html [hunk 2]
```diff
@@ -206,6 +236,45 @@ <h6>UNIVERSITY</h6>
     {{/menteeProfiles}}     
 </script>
 
+<script type="text/html" id="template-profile-modal-content">
```
Comment (YoshithaRathnayake) [inline scholarx/archive/index.html:232]: Added from this line

**Comment by Madhawa97** [comment]
Well, It's not a good practice to format code. Makes the reviewing lot harder.
Can you undo it?

**Comment by YoshithaRathnayake** [comment]
> Well, It's not a good practice to format code. Makes the reviewing lot harder.
Can you undo it?

Ok then I will make it on a new pr because I can't undo them in vs code, after closing can't undo the changes. I format the edited code @Madhawa97 but it's formatted all

File: scholarx/archive/index.html [hunk 3]
```diff
@@ -30,153 +30,183 @@
...
+                <div id="showMentors" class="mentors-table">
+                    <table>
+                        <thead>
+                            <tr>
+                                <th></th>
+                                <th>Name</th>
+                                <th>Details</th>
+                            </tr>
+                        </thead>
+                        <tbody id="mentorProfiles" class="profile-model">
```
Comment (YoshithaRathnayake) [inline scholarx/archive/index.html]: Added a class as profile model

File: scholarx/archive/index.html [hunk 4]
```diff
@@ -30,153 +30,183 @@
...
+                    <table>
+                        <thead>
+                            <tr>
+                                <th></th>
+                                <th>Name</th>
+                                <th>University</th>
+                                <th>Mentor</th>
+                            </tr>
+                        </thead>
+                        <tbody id="menteeProfiles" class="profile-model">
```
Comment (YoshithaRathnayake) [inline scholarx/archive/index.html]: here too

File: scholarx/archive/styles.css [hunk 5]
```diff
@@ -76,3 +76,7 @@ hr {
   text-align: left;
   padding: 8px;
 }
+
+.profile-model:hover {
```
Comment (YoshithaRathnayake) [inline scholarx/archive/styles.css]: Add this css that's it I didn't edit other lines

**Comment by Madhawa97** [comment]
You can revert to a previous commit or get the code from a previous commit (before you format) and add your changes again.

**Comment by YoshithaRathnayake** [comment]
> You can revert to a previous commit or get the code from a previous commit (before you format) and add your changes again.

I'll try

**Review by anjula-sack — CHANGES_REQUESTED** [review]
can you resolve the conflicts? @YoshithaRathnayake

**Comment by YoshithaRathnayake** [comment]
> can you resolve the conflicts? @YoshithaRathnayake

@anjula-sack there are lot of conflicts, can you tell me how to resolve them locally?

**Comment by YoshithaRathnayake** [comment]
> can you resolve the conflicts? @YoshithaRathnayake

Sure @anjula-sack

**Comment by YoshithaRathnayake** [comment]
@Madhawa97 @anjula-sack Please review this now, @janithlahirukariyawasam helped me to do this

**Review by anjula-sack — CHANGES_REQUESTED** [review]
can you resolve the conflicts? @YoshithaRathnayake

**Comment by Piumal1999** [comment]
@YoshithaRathnayake Thanks for the pull request, but I'm closing this since the requested feature has already been implemented with #1525.

**Pull request closed without merging**